primitive-types/impls/serde/src/serialize.rs

```diff
@@ -36,6 +62,49 @@ fn to_hex<'a>(v: &'a mut [u8], bytes: &[u8], skip_leading_zero: bool) -> &'a str
 	unsafe { std::str::from_utf8_unchecked(&v[0..idx]) }
 }
 
+/// Decode given hex string into a vector of bytes.
+///
+/// Returns an error if the string is not prefixed with `0x`
+/// or non-hex characters are present.
+pub fn from_hex(v: &str) -> Result<Vec<u8>, String> {
```

Comment:
is not symmetric with to_hex which can generate strings both with `0x` and without?

Reply:
Ah ok, I've misread the implementation